src/lib/registration/addon.rb

```diff
@@ -125,23 +169,46 @@ def create_addon_with_deps(root)
     # @param pure_addon [SUSE::Connect::Product] a pure add-on from the registration server
     def initialize(pure_addon)
       @pure_addon = pure_addon
-      @children = []
     end
 
     # is the add-on selected
-    # @return [Boolean] true if the add-on is selectec
+    # @return [Boolean] true if the add-on is selected
     def selected?
       Addon.selected.include?(self)
     end
 
+    # is the add-on auto_selected
+    # @return [Boolean] true if the add-on is auto_selected
+    def auto_selected?
+      Addon.auto_selected.include?(self)
+    end
+
     # select the add-on
     def selected
-      Addon.selected << self unless selected?
+      return if selected?
+
+      Addon.selected << self
+      Addon.reset_auto_selected
     end
 
     # unselect the add-on
     def unselected
-      Addon.selected.delete(self) if selected?
+      return unless selected?
+
+      Addon.selected.delete(self)
+      Addon.reset_auto_selected
+    end
+
+    # returns status of addon. Potential statuses are :registered, :selected, :auto_selected,
+    # :available and :none.
+    # @return [Symbol]
+    def status
+      return :registered if registered?
+      return :selected if selected?
+      return :auto_selected if auto_selected?
+      return :available if available?
+
+      :none
```

Comment:
Maybe `:unknown` would be a bit better here? An addon always has a state, in this case we just do not know which one.

I guess this can only happen if you pass some addon created from outside, which the code does not know. An then `:unknown` sounds good...

Reply:
@lslezak well, I am thinking about good name. This state is in fact `:unavailable` - so not selected and also not available on scc. But I do not like it much.

Reply:
If it is not available in SCC then we do not know it and thus `:unknown` still makes sense. Probably not the best name, but IMHO still better than `:none`.

And this is just a theoretical case, normally we should never reach this pont. Maybe we should log a warning if it actually happens?

Reply:
make sense